### ISSUE TITLE
Frame rate hit / graphics stuttering/tearing

### ISSUE THREAD

**Issue by Camisade** [lekeno/edr#219]
I'm a new EDR user, just granted credentials. My processor is an Intel Core i7-4390K @ 3.4GHz. Graphics card is nVidia GTX 1080. BenQ monitor running at 60 Hz. Elite Dangerous runs at 60FPS all the time (even during packhound launch!). Elite Dangerous resolution is set to 3840x2160, 60Hz, windowed mode (because I like to run Discord and other apps on both primary and secondary monitors -- but none are CPU intensive enough that they cause any FPS drop when playing Elite Dangerous without EDR loaded).  I'm running the latest production nVidia drivers.

When EDR is loaded with EDMC, FPS drops down to 55-57 FPS (as measured within Elite Dangerous), and graphics begin stuttering, with some slight tearing. When I remove EDR, the graphics issues go away again.

Anything I can do to help you further diagnose this problem?

**Comment by Camisade** [comment]
Add'l note: When EDR puts an overlay up, FPS drops as low as 32-34.

**Comment by lekeno** [comment]
The overlay is from https://github.com/inorton/EDMCOverlay/
Perhaps, inorton would have some ideas to improve its performance...

The overlay does have an impact but cutting in half the FPS is extreme (I haven't heard many folks with that sort of issue)...

In the meantime, you might want to try a few things:
 - Borderless vs. Fullscreen vs. Windowed
 - Uncheck VSync in Elite Dangerous' Graphics options
 - Increase the frame rate limit beyond 60hz in Elite Dangerous' Graphics options.
 - Disable the overlay by going into EDMC's settings, EDR tab and uncheck Overlay. Instead, check the box on the right hand side of the EDR status line in EDMC to reveal an alternative UI. Optionally, go to EDMC's settings, appearance and check Always visible, and Transparent.

**Comment by Camisade** [comment]
Update on this: Unchecking VSync in the Elite Dangerous graphics options, after changing to a faster monitor (144Hz with G-Sync functionality) (mostly) did the trick. There is some minor stuttering of the display when EDR writes to the overlay, but it's minor.

Tl;dr version:
* Nothing I tried with my original monitor (a BenQ PD32U) seemed to help.
* After changing to a faster gaming monitor (Acer XG27), the stuttering was still pronounced and frame rates during the overlay would drop from 130-144 down to 80-90, then come back up after the overlay disappeared.
* I first tried getting rid of the overlay, and that helped, but I really wanted the OSD rather than having to check my second monitor for the EDR report.
* U re-enabled the OSD and unchecked the VSync. Voila! Problem (mostly) solved.

Thanks for the help.
On 1/20/2019 9:41:00 AM, LeKeno <notifications@github.com> wrote:
The overlay is from https://github.com/inorton/EDMCOverlay/ [https://github.com/inorton/EDMCOverlay/]
Perhaps, inorton would have some ideas to improve its performance...
The overlay does have an impact but cutting in half the FPS is extreme (I haven't heard many folks with that sort of issue)...
In the meantime, you might want to try a few things:
* Borderless vs. Fullscreen vs. Windowed
* Uncheck VSync in Elite Dangerous' Graphics options
* Increase the frame rate limit beyond 60hz in Elite Dangerous' Graphics options.
* Disable the overlay by going into EDMC's settings, EDR tab and uncheck Overlay. Instead, check the box on the right hand side of the EDR status line in EDMC to reveal an alternative UI. Optionally, go to EDMC's settings, appearance and check Always visible, and Transparent.
